src/transaction.js

```diff
@@ -846,27 +1057,40 @@ Transaction.prototype.__toBuffer = function (buffer, initialOffset, __allowWitne
       joinsplit.macs.forEach(function (nullifier) {
         writeSlice(nullifier)
       })
-      writeCompressedG1(joinsplit.zproof.gA)
-      writeCompressedG1(joinsplit.zproof.gAPrime)
-      writeCompressedG2(joinsplit.zproof.gB)
-      writeCompressedG1(joinsplit.zproof.gBPrime)
-      writeCompressedG1(joinsplit.zproof.gC)
-      writeCompressedG1(joinsplit.zproof.gCPrime)
-      writeCompressedG1(joinsplit.zproof.gK)
-      writeCompressedG1(joinsplit.zproof.gH)
+      if (this.version >= Transaction.ZCASH_SAPLING_VERSION) {
+        writeSlice(joinsplit.zkproof.sA)
+        writeSlice(joinsplit.zkproof.sB)
+        writeSlice(joinsplit.zkproof.sC)
+      } else {
+        writeCompressedG1(joinsplit.zkproof.gA)
+        writeCompressedG1(joinsplit.zkproof.gAPrime)
+        writeCompressedG2(joinsplit.zkproof.gB)
+        writeCompressedG1(joinsplit.zkproof.gBPrime)
+        writeCompressedG1(joinsplit.zkproof.gC)
+        writeCompressedG1(joinsplit.zkproof.gCPrime)
+        writeCompressedG1(joinsplit.zkproof.gK)
+        writeCompressedG1(joinsplit.zkproof.gH)
+      }
       joinsplit.ciphertexts.forEach(function (ciphertext) {
         writeSlice(ciphertext)
       })
-    })
+    }, this)
     if (this.joinsplits.length > 0) {
       writeSlice(this.joinsplitPubkey)
       writeSlice(this.joinsplitSig)
     }
   }
 
+  if (coins.isZcash(this.network) &&
+    this.version >= Transaction.ZCASH_SAPLING_VERSION &&
+    this.vShieldedSpend.length + this.vShieldedOutput.length > 0) {
+    writeSlice(this.bindingSig)
+  }
+
   // avoid slicing unless necessary
   if (initialOffset !== undefined) return buffer.slice(initialOffset, offset)
-  return buffer
+  // TODO: we are doing some miscalculation somewhere, we shouldn't have to slice the final buffer
```

Comment:
All TODOs should come with a JIRA link.

Reply:
Create an issue in github and add it to the comment here and then I'm good with resolving this.